src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
 Programmable Physical Environments
 ==================================
 
+Overview
+--------
+
+Physical things includes nearly everything that will appear in life of human beings. When it comes to programmable physical things, the term should refer to a kind of physical system which might be a combination of software and hardware which can respond to the analog world. When we say 'environment', we focus more on the interactive function of this physical system with human beings. Now we can derive a definition for programmable physical environment, which is an interactively programmable physical system that can sense and respond to the real world.
+
+Programmable physical environment has a broad definition and it is used in various fields. The usage included educational use, virtual/augmented reality, factory manufacture, internet of things, building architecture, etc. The term still has a long way for improvement. In the future, programmable physical environment may become more common in the society as a combination of various technology systems.
```

Comment:
- "The term still has a long way for improvement" -- terms aren't improved. 
- Start this paragraph with "This is an active research area with many applications, including in education, ..."